nvidia-nano/GiveMeWES.yaml

```diff
@@ -0,0 +1,597 @@
+---
+- name: Check Jetpack version
+  hosts: all
+  vars_prompt:
+
+    - name: Connect Drive
+      prompt: Please connect your usb drive to the nano. When ready press enter
+      private: no
+
+  tasks:
+    - name: Update repositories cache
+      ansible.builtin.apt:
+        update_cache: yes
+
+    - name: Register jetpack version
+      command: apt-cache show nvidia-jetpack
+      register: jetpackVer
+
+    - name: Test for Jetpack 4.4.1
+      fail: msg="The wrong version of Jetpack was flashed. I want jetpack 4.4.1"
+      when: not ('4.4.1' in jetpackVer.stdout)
+
+- name: Set Up Extra Drive
+  hosts: all
+  tasks:
+    - name: Clear the current partition table and create GPT table
+      shell: echo -e "g\nw" | fdisk --wipe always --wipe-partitions always /dev/sda #Can't use parted module because it has issues clearing partitions in use
+      args:
+        executable: /bin/bash
+
+    - name: Make the SWAP partition (16GB SWAP) #Can't create swap partition with parted module so use shell command
+      shell: echo -e "n\n1\n2048\n+16G\nt\n19\nw" | fdisk --wipe always --wipe-partitions always /dev/sda
+      args:
+        executable: /bin/bash
+
+    - name: Make the overlayfs 32GB
+      community.general.parted:
+        device: /dev/sda
+        number: 2
+        state: present
+        label: gpt
+        name: system-data
+        part_start: 33556480s #s = sectors
+        part_end: 100665343s 
+        fs_type: ext4
+
+    - name: Make the plugin-data (the rest of the space)
+      community.general.parted:
+        device: /dev/sda
+        number: 3
+        state: present
+        label: gpt
+        name: plugin-data
+        part_start: 100665344s  
+        part_end: 100%
+        fs_type: ext4
+
+    - name: Create a ext4 filesystem on /dev/sda2
+      community.general.filesystem:
+        fstype: ext4
+        dev: /dev/sda2
+
+    - name: Create a ext4 filesystem on /dev/sda3
+      community.general.filesystem:
+        fstype: ext4
+        dev: /dev/sda3
+
+    - name: Unmount /dev/sda1 #Unmount this partition so that mkswap can run
+      ansible.posix.mount:
+        path: /dev/sda1
+        state: unmounted
+
+    - name: Turn on the swap
+      shell: mkswap /dev/sda1 -L ext-swap
+      args:
+        executable: /bin/bash
+
+    - name: Put the swap in the startup partition file #table structure: /dev/sda1 swap swap defaults,nofail 0 0
+      ansible.posix.mount:
+        path: swap
```

Comment:
need to confirm this looks right in the /etc/fstab file (matching it done by hand)

Reply:
done, switched to fstab